receiver/kafkareceiver/azureresourcelogs_unmarshaler.go

```diff
@@ -0,0 +1,196 @@
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/relvacode/iso8601"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	conventions "go.opentelemetry.io/collector/semconv/v1.13.0"
+)
+
+// Constants for OpenTelemetry Specs
+const receiverScopeName = "otelcol/" + typeStr
+
+// Constants for Azure Log Records
+const azureCategory = "azure.category"
+const azureCorrelationID = "azure.correlation.id"
+const azureDuration = "azure.duration"
+const azureIdentity = "azure.identity"
+const azureOperationName = "azure.operation.name"
+const azureOperationVersion = "azure.operation.version"
+const azureProperties = "azure.properties"
+const azureResourceID = "azure.resource.id"
+const azureResultType = "azure.result.type"
+const azureResultSignature = "azure.result.signature"
+const azureResultDescription = "azure.result.description"
+const azureTenantID = "azure.tenant.id"
+
+type azureResourceLogsUnmarshaler struct{}
+
+// azureRecords represents an array of Azure log records
+// as exported via an Azure Event Hub
+type azureRecords struct {
+	Records []azureLogRecord `json:"records"`
+}
+
+// azureLogRecord represents a single Azure log following
+// the common schema:
+// https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/resource-logs-schema
+type azureLogRecord struct {
+	Time              string       `json:"time"`
+	ResourceID        string       `json:"resourceId"`
+	TenantID          *string      `json:"tenantId"`
+	OperationName     string       `json:"operationName"`
+	OperationVersion  *string      `json:"operationVersion"`
+	Category          string       `json:"category"`
+	ResultType        *string      `json:"resultType"`
+	ResultSignature   *string      `json:"resultSignature"`
+	ResultDescription *string      `json:"resultDescription"`
+	DurationMs        *string      `json:"durationMs"`
+	CallerIPAddress   *string      `json:"callerIpAddress"`
+	CorrelationID     *string      `json:"correlationId"`
+	Identity          *interface{} `json:"identity"`
+	Level             *string      `json:"Level"`
+	Location          *string      `json:"location"`
+	Properties        *interface{} `json:"properties"`
+}
+
+func newAzureResourceLogsUnmarshaler() LogsUnmarshaler {
+	return azureResourceLogsUnmarshaler{}
+}
+
+func (r azureResourceLogsUnmarshaler) Unmarshal(buf []byte) (plog.Logs, error) {
+	l := plog.NewLogs()
+
+	var azureLogs azureRecords
+	decoder := jsoniter.NewDecoder(bytes.NewReader(buf))
+	if err := decoder.Decode(&azureLogs); err != nil {
+		return l, err
+	}
+
+	resourceLogs := l.ResourceLogs().AppendEmpty()
+	scopeLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	scopeLogs.Scope().SetName(receiverScopeName)
+	scopeLogs.Scope().SetVersion("0.70.0")
+	logRecords := scopeLogs.LogRecords()
+
+	resourceID := ""
+	for _, azureLog := range azureLogs.Records {
+		resourceID = azureLog.ResourceID
+		nanos, err := asTimestamp(azureLog.Time)
+		if err != nil {
+			continue
```

Comment:
Is it worth adding a debug statement here in the event a customer needs to debug this?

Reply:
Yes, I added it to the Event Hub Unmarshaler already in that PR.